src/sys/windows.rs

```diff
@@ -735,6 +736,52 @@ pub(crate) fn from_in6_addr(addr: in6_addr) -> Ipv6Addr {
     Ipv6Addr::from(*unsafe { addr.u.Byte() })
 }
 
+/// This type is not yet in winapi.
+#[repr(C)]
+#[allow(non_camel_case_types)]
+struct sockaddr_un {
+    pub sun_family: sa_family_t,
+    pub sun_path: [u8; 108],
+}
+
+#[allow(unused_unsafe)] // TODO: replace with `unsafe_op_in_unsafe_fn` once stable.
+pub(crate) fn unix_sockaddr(path: &Path) -> io::Result<SockAddr> {
+    unsafe {
+        SockAddr::init(|storage, len| {
+            // Safety: `SockAddr::init` zeros the address, which is a valid
+            // representation.
+            let storage: &mut crate::sys::sockaddr_un = unsafe { &mut *storage.cast() };
+            let len: &mut socklen_t = unsafe { &mut *len };
+
+            let bytes = path
+                .to_str()
+                .ok_or_else(|| {
+                    io::Error::new(io::ErrorKind::InvalidInput, "path must be valid UTF-8")
+                })?
+                .as_bytes();
+
+            if bytes.len() > storage.sun_path.len() {
+                return Err(io::Error::new(
+                    io::ErrorKind::InvalidInput,
+                    "path must be shorter than SUN_LEN",
+                ));
+            }
+
+            storage.sun_family = crate::sys::AF_UNIX as sa_family_t;
+            storage.sun_path[..bytes.len()].copy_from_slice(bytes);
```

Comment:
Can you mention `storage.sun_path` is zero filled and thus already null terminated, same as https://github.com/rust-lang/socket2/blob/1c67209c2eda5708b3093900a59befbc4a981b13/src/sys/unix.rs#L477-L478

Reply:
Done.